cmd/lift/lift.go

```diff
@@ -18,30 +28,96 @@ func lift(chainFile string, inFile string, outFile string) {
 	}
 	tree := interval.BuildTree(chainIntervals)
 	out := fileio.EasyCreate(outFile)
+	defer out.Close()
+
+	un := fileio.EasyCreate(unMapped)
+	defer un.Close()
+
+	var records []*fasta.Fasta
+	var currVcf *vcf.Vcf
+	var a, b float64
+
+	if faFile != "" {
+		if !vcf.IsVcfFile(inFile) {
+			log.Fatalf("Fasta files should be provided when lifting VCF files. inFile is not a valid VCF file and must end in either .vcf or .vcf.gz.")
+		}
+		records = fasta.Read(faFile)
+	}
+
 	//second task, read in intervals, find chain, and convert to new interval
 	inChan := interval.GoReadToLiftChan(inFile)
 	var overlap []interval.Interval
 	for i := range inChan {
-		overlap = interval.Query(tree, i, "any")
+		overlap = interval.Query(tree, i, "any")//TODO: verify proper t/q contains
 		if len(overlap) > 1 {
-			log.Fatalf("Multiple overlaps???")
+			fmt.Fprintf(un, "Record below maps multiple chains:\n")
+			i.WriteToFileHandle(un)
+		} else if len(overlap) == 0 {
+			fmt.Fprintf(un, "Record below has no ortholog in new assembly:\n")
+			i.WriteToFileHandle(un)
+		} else if !minMatchPass(overlap[0].(*chain.Chain), i, minMatch) {
+			a, b = interval.MatchProportion(overlap[0].(*chain.Chain), i)
+			fmt.Fprintf(un, "Record below fails minMatch with a proportion of %f:\n", numbers.MinFloat64(a, b))
+			fmt.Fprintf(un, "Here's the corresponding chain: Score: %d.\n", overlap[0].(*chain.Chain).Score)
+			i.WriteToFileHandle(un)
+		} else {
+			//DEBUG: interval.PrettyPrint(i)
+			i.UpdateLift(interval.LiftIntervalWithChain(overlap[0].(*chain.Chain), i))//now i is 1-based and we can assert VCF.
+			//DEBUG: interval.PrettyPrint(i)
+			//special check for lifting over VCF files
+			if faFile != "" {
+				//faFile will be given if we are lifting over VCF data.
+				currVcf = i.(*vcf.Vcf)
+				if utf8.RuneCountInString(currVcf.Ref) > 1 {
```

Comment:
Do we also need to check Alt?  I see the log fatal says we don't handle indels (which is fine), but I am picture the ref is A and the alt is ACGT, which I don't think get caught by this check, even though it is an indel